### ISSUE TITLE
How to only predict class labels of known ground truth boxes?

### ISSUE THREAD

**Issue by simoneriggi** [ultralytics/ultralytics#17079]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Dear all,
I have a trained YOLO v8 model and I want to use it on a new dataset (for which I already have the ground truth object boxes, provided externally) only to predict the known ground truth box class and score.
Would it be possible? If not, what should I modify in the code to enable this feature?

Thanks a lot for your help.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @simoneriggi, thank you for your interest in Ultralytics 🚀! We recommend looking through the documentation for a wealth of information on Python and CLI usage examples, and to see if any of your questions may already be answered.

If this is a 🐛 Bug Report, please provide a minimum reproducible example to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our tips for best training results.

Join the Ultralytics community where it suits you best. For real-time chat, head to our community Discord 🎧. Prefer in-depth discussions? Check out our Discourse forum. Or dive into threads on our Subreddit to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package in a **Python>=3.8** environment with **PyTorch>=1.8** to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments, including Notebooks with free GPU access, Google Cloud Deep Learning VM, Amazon Deep Learning AMI, or Docker images.

## Status

Check our continuous integration status to ensure tests verify the correct operation of all YOLO modes and tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response, but rest assured that an Ultralytics engineer will assist you soon. 😊

**Comment by Y-T-G** [comment]
If the class ids match between your training dataset and new dataset, then you can pass the ids as list to `classes` argument in `val()`

`model.val(..., classes=[1,2,3])`

**Comment by Y-T-G** [comment]
Yeah, the best thing is to fix the labels by running find and replace.

I have a script for it.

https://gist.github.com/Y-T-G/6ad550e03822cbd9b0722c83e3dfe2cb

**Comment by Y-T-G** [comment]
You could map the classes that don't exist in the original dataset to None, which would remove them from the second dataset in the script. And you can map multiple classes to the same original classes too. I don't see the issue.

**Comment by Y-T-G** [comment]
> Instead, I need to run the model prediction on known bounding boxes and see how the model actually label them.

I don't understand what you mean by this. The model has no foreknowledge of the ground truth based on which it can exclude certain classes a priori.
